lib/data/external/AwsClient.js

```diff
@@ -356,6 +358,32 @@ class AwsClient {
             return callback();
         });
     }
+    copyObject(request, sourceObjectKey, log, callback) {
+        // for backwards compatibility
+        const destBucketName = request.bucketName;
+        const destObjectKey = request.objectKey;
+        const awsKey = this._createAwsKey(destBucketName, destObjectKey,
+          this._bucketMatch);
+        const metadataDirective = request.headers['x-amz-metadata-directive'];
+        const metaHeaders = trimXMetaPrefix(getMetaHeaders(request.headers));
+        this._client.copyObject({
+            Bucket: this._awsBucketName,
+            Key: awsKey,
+            CopySource: `${this._awsBucketName}/${sourceObjectKey}`,
+            Metadata: metaHeaders,
+            MetadataDirective: metadataDirective,
+        }, err => {
+            if (err) {
+                logHelper(log, 'error', 'error from data backend on ' +
+                'copyObject', err, this._dataStoreName);
+                return callback(errors.InternalError
```

Comment:
As discussed, if it's access denied we should probably pass that through so requester knows that no access on either source or destination bucket. Also, we need a test where the access key pair for the source bucket and destination bucket are different and there is an access denied.